contracts/AnyswapV4CallProxy.sol

```diff
@@ -140,8 +187,10 @@ contract AnyCallProxy is Whitelistable {
         address[] memory callbacks,
         uint256[] memory nonces,
         uint256 toChainID
-    ) external onlyFromWhitelist(msg.sender) onlyToWhitelist(msg.sender, toChainID, to) {
-        emit LogAnyCall(msg.sender, to, data, callbacks, nonces, cID, toChainID);
+    ) external onlyWhitelisted(msg.sender, toChainID, to) {
+        require(toChainID != block.chainid, "AnyCall: FORBID");
```

Comment:
Calls to the same `chainid` as the source shouldn't be a thing. If the caller can't interact with the contract normally it shouldn't be done using anycall as a proxy. Also this allows integrators to do checks such as:
`require(msg.sender == ANYCALL)`, and not be subject to attacks where an attacker calls the contract on the same chain using anycall as a proxy.